src/main/java/hudson/plugins/spotinst/cloud/BaseSpotinstCloud.java

```diff
@@ -587,25 +591,31 @@ protected Integer getNumOfExecutors(String instanceType) {
         Integer retVal;
 
         Integer globalOverrideExecutorsNumber = getExecutorsFromGlobalOverride();
+        boolean isOneOffNodesEnabled          = getIsOneOffNodesEnabled();
 
-        if (globalOverrideExecutorsNumber != null) {
-            LOGGER.debug(String.format("Overriding executors for instance type %s to be %s", instanceType,
-                                       globalOverrideExecutorsNumber));
-            retVal = globalOverrideExecutorsNumber;
+        if (isOneOffNodesEnabled) {
+            retVal = 1;
         }
         else {
-            Integer defaultNumberOfExecutors = getDefaultExecutorsNumber(instanceType);
-
-            if (defaultNumberOfExecutors != null) {
-                retVal = defaultNumberOfExecutors;
+            if (globalOverrideExecutorsNumber != null) {
```

Comment:
AwsSpotinstCloud should not override it any more, instead override other method that called : getOverridedExecuter which return const NO_OVERRIDE = -1, in aws it return the overrided value or -1 if no override

Reply:
done